pysteps/blending/steps.py

```diff
@@ -557,22 +572,19 @@ def forecast(
         fft,
     )
 
-    # 2.2 If necessary, decompose (NWP) model forecasts and stack cascades
-    (
-        precip_models_cascade,
-        mu_models,
-        sigma_models,
-        precip_models_pm,
-    ) = _compute_cascade_decomposition_nwp(
-        precip_models, bp_filter, decompositor, recompositor, fft, domain
-    )
+    # 2.2 If necessary, recompose (NWP) model forecasts
```

Comment:
Does this mean that users who provided the decomposed NWP will then have it recomposed again? Why is this only done if the ndim =/= 4? If this check of ndims != 4 is to assume that you are in the decomposed case, I feel this is not very elegant. Would it not better be to have an extra variable for the user to finetune their call of the function where they have to determine this with the default being the old approach?

Reply:
Well yes, but unfortunately this is how it was done before. Before these changes, both supplying decomposed and not decomposed data was already possible, and the check to see which of the 2 was supplied, was this ndim check.

An extra variable would have my preference as well, but that would be a breaking change with what it was before. Maybe you can take this up with your refactoring?

Reply:
Good point, I did not know this was already possible! I will take this up as an issue for my refactoring